src/signer/interface.ts

```diff
@@ -14,4 +15,22 @@ export abstract class SignerInterface extends Provider {
   public abstract override addTransaction(
     transaction: Transaction
   ): Promise<AddTransactionResponse>;
+
+  /**
+   * Sign an JSON object with the starknet private key and return the signature
+   *
+   * @param json - JSON object to be signed
+   * @returns the signature of the JSON object
+   * @throws {Error} if the JSON object is not a valid JSON
+   */
+  public abstract signMessage(typedData: TypedData): Promise<Signature>;
```

Comment:
More food for thought, but with account abstraction it is possible to have an account controlled by multiple keys and that all keys must sign an off-chain message if it is to be valid. In that case the method would have to return an array of signatures. Maybe that is something we should anticipate now.

Note that the Account Contract interface in Argent or OpenZeppelin (which can validate signatures with `is_valid_signature` ) already supports signatures as an array.